### PR TITLE
Add DEXSeq and Subread_to_DEXSeq to resolwebio/rnaseq

### PR DESCRIPTION
**Pull request by jvrakor**
@jkokosar @mzganec 

This image is built on docker.io/resolwebio/base:ubuntu-18.04 which is build on docker.io/resolwe/base:ubuntu-18.04. These two docker images do not yet exist. Both are built on Ubuntu 18.04.

### CODE REVIEW

File: resolwe_bio/docker_images/rnaseq/README.md [hunk 1]
```diff
@@ -9,11 +9,14 @@ Included bioinformatics tools:
 * Cufflinks (v2.2.1)
 * Cutadapt (v1.14)
 * DESeq2 (v1.16.1)
+* DEXSeq (v1.24.2)
 * FastQC (v0.11.5)
 * HISAT2 (v2.1.0)
 * HTSeq (v0.9.1)
+* HTSeq (python2) (v0.6.1p1)
```
Comment (mstajdohar) [inline resolwe_bio/docker_images/rnaseq/README.md]: Why do we need this?

**Comment by jkokosar** [comment]
Current build on dockerhub for this image is failing with:
```[91mTHESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
    cutadapt==1.14 from https://pypi.python.org/packages/6d/60/ff2013968bb807af8d999c317a40680a363ff30c005db6657f58654d79ad/cutadapt-1.14-cp36-cp36m-manylinux1_x86_64.whl#md5=a31c6b1e3332b1b1ce54ca96e2e91f20 (from -r /var/cache/build/packages-python3.txt (line 1)):
        Expected sha256 9c1423123e764bc95e96d196ee8025a5ec213ffab218fc09088bdd2b577b3a04
             Got        a9903b9288956c6e8c4652670efa4eda40b258f29f00f3b85ae6b16d80b022c8

    HTSeq==0.9.1 from https://pypi.python.org/packages/eb/70/0f23df31b3587f1219758339723c3d442d7fd0a00dd0857846fe11ed7b09/HTSeq-0.9.1-cp36-cp36m-manylinux1_x86_64.whl#md5=5afda8bdabca11cc77fb93dae77bb617 (from -r /var/cache/build/packages-python3.txt (line 2)):
        Expected sha256 f61cd2a19aa12514b3b9d4fa0725cbf8d34b06313da6c468d38cfe9aef84b350
             Got        04a7e6987c74769bae40e1da51ebdb22861d2dcd6edca71828a4527b98b847db

    pyarrow==0.7.1 from https://pypi.python.org/packages/2d/2b/52d3f5828cf6253f36eed849e4b15e11a0d0ed258a9cd7f38c3867b02024/pyarrow-0.7.1-cp36-cp36m-manylinux1_x86_64.whl#md5=acea1613e2d9c491736e533bd68bf4ae (from -r /var/cache/build/packages-python3.txt (line 3)):
        Expected sha256 7908c71b0109d6f622436bdf719cde5b749287a697134d23040e35d32c28a313
             Got        2c4b8120d8a45896626fa874092f7a20df24f1655e2da5a6a2b045bce03b11a1

    pysam==0.12.0.1 from https://pypi.python.org/packages/50/28/f3a3989a9a34fa6c84fdba3d80b4b1624b2c253d7906a13638152b20555b/pysam-0.12.0.1-cp36-cp36m-manylinux1_x86_64.whl#md5=26c61980d309ac339dd4ef21e1b1d58e (from -r /var/cache/build/packages-python3.txt (line 4)):
        Expected sha256 a99cd626ff0fdc09b68a1cc9cc9551f4dc3c4cd49872583de1a2f39e69520923
             Got        f1cd051d474a72079dbeaa3d6cb1b26dac005af8e4cf71b4dddad7c66b08ce26```

Test the build of this image locally to fix the hash values as required.

File: resolwe_bio/docker_images/rnaseq/README.md (hunk 1 above)
Comment (jvrakor) [reply resolwe_bio/docker_images/rnaseq/README.md]: There are python2 scripts in DEXSeq and Subread_to_DEXSeq that use HTSeq. These scripts are part of the packages and should not be tempered with. We could ask the authors to change them or make pull requests to their git directories.

**Comment by mzganec** [comment]
Python packages fail to build due to the recent bump of `resolwebio/base` Docker image to 17.10 https://github.com/genialis/resolwe-bio/commit/bd85d7d009a87e74818a380d1e5579c2620786ec

**Comment by jvrakor** [comment]
I don't see how though. `pip install <package==version>` should install the same package regardless of the system and the same goes for `pip3 install`

**Comment by kostko** [comment]
This is not true. If the Python version changes (e.g. from 3.5 to 3.6), then `pip` may install different packages as binary packages are built per-Python-version.

File: resolwe_bio/docker_images/rnaseq/packages-python3.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
-cutadapt==1.14 --hash=sha256:9c1423123e764bc95e96d196ee8025a5ec213ffab218fc09088bdd2b577b3a04
-HTSeq==0.9.1 --hash=sha256:f61cd2a19aa12514b3b9d4fa0725cbf8d34b06313da6c468d38cfe9aef84b350
-pyarrow==0.7.1 --hash=sha256:7908c71b0109d6f622436bdf719cde5b749287a697134d23040e35d32c28a313
-pysam==0.12.0.1 --hash=sha256:a99cd626ff0fdc09b68a1cc9cc9551f4dc3c4cd49872583de1a2f39e69520923
+cutadapt==1.14 --hash=sha256:a9903b9288956c6e8c4652670efa4eda40b258f29f00f3b85ae6b16d80b022c8
```
Comment (mzganec) [inline resolwe_bio/docker_images/rnaseq/packages-python3.txt]: While you're at it, check the latest version of all python packages installed in the RNA-seq docker image. All packages except HTSeq can be updated. This should become our standard practice to keep the Docker images well maintained.

**Comment by mzganec** [comment]
Building `HTSeq` reports an error `Failed building wheel for htseq`. The same goes for `pysam`.

Use the latest version of `HTSeq` - we want the latest version of all bioinformatics tools in this Docker image. For older versions we have the `resolwebio/rnaseq:legacy` Docker image.

Installation of `HTSeq` with `pip` makes `htseq-count` available on `$PATH`. Having two versions of `HTSeq` installed in the same Docker image makes it difficult to control which version is called in bash from [htseq-count process](https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/processes/expression/htseq_count.yml#L147). For this reason I dislike the idea of having both versions installed in the same Docker image.

I see two solutions: (1) The [script](https://github.com/vivekbhr/Subread_to_DEXSeq/blob/master/dexseq_prepare_annotation2.py) that requires the Python 2 version of `HTSeq` is really easy to update for Python3. The `DEXSeq` manual even provides a detailed explanation of this script in case you want to modify it or write your own. Only two changes are required: `raise ValueError` needs to be called with parentheses and `xrange` needs to be replaced with `range`. You can also make a pull request in that repository. (2) Use a separate Docker image.

File: docs/CHANGELOG.rst [hunk 3]
```diff
@@ -33,8 +33,9 @@ Added
 * Add ``prepare-geo-chipseq`` and ``prepare-geo-rnaseq`` processes that
   produce a tarball with necessary data and folder structure for GEO
   upload
-* Add ``BBMap``, ``Trimmomatic``, ``Subread``, and ``Salmon`` tool
-  to ``resolwebio/rnaseq`` Docker image
+* Add ``BBMap``, ``Trimmomatic``, ``Subread``, ``Salmon``, and
+  ``dexseq_prepare_annotation2`` tools and ``DEXSeq`` and
+  ``loadSubread`` R libraries to ``resolwebio/rnaseq`` Docker image
 * Add ``Salmon`` transcript quantification tool to
```
Comment (mzganec) [inline docs/CHANGELOG.rst]: Remove this entry because it's a duplication.

**Pull request merged**